templates/flavors/README.md

```diff
@@ -65,9 +65,9 @@ worker-templates:
     AZURE_RESOURCE_GROUP: test-resource-group-name
     CONTROL_PLANE_MACHINE_COUNT: "1"
     KUBERNETES_VERSION: v1.22.1
-    AZURE_CONTROL_PLANE_MACHINE_TYPE: Standard_D2s_v3
+    AZURE_CONTROL_PLANE_MACHINE_TYPE: Standard_B2s
```

Comment:
Should we also effectively recommend that users start with `Standard_B2s`? It seems to handle our e2e workloads, but I don't know whether real-world use might find it lacking.

Reply:
I'm not sure that the current existence of `Standard_D2s_v3` is anything more than a "here's a known-working, barely operational example" for a VM SKU.

In other words, we have a long history of including in our reference configs the VM SKU that we use for testing — which includes in its positive criteria "cost-efficient" — (AFAICT this is the reason we use `Standard_D2s_v3`, not that it's actually a SKU we would recommend for real-world production use) so I don't think this amounts to a significant change.

Reply:
Makes sense to me, thanks for the thoughtful comment.

Also, it's really in the Cluster API Book [Quick Start](https://cluster-api.sigs.k8s.io/user/quick-start.html#required-configuration-for-common-providers) that we say anything that could be interpreted as a recommendation:

```shell
# Select VM types.
export AZURE_CONTROL_PLANE_MACHINE_TYPE="Standard_D2s_v3"
export AZURE_NODE_MACHINE_TYPE="Standard_D2s_v3"
```

and that's not going to change with this PR. So I guess this doesn't rock the boat that badly.